MagazineLayout/LayoutCore/ModelState.swift

```diff
@@ -377,11 +379,9 @@ final class ModelState {
       sectionModels[indexPath.section].updateItemHeight(
         toPreferredHeight: preferredHeight,
         atIndex: indexPath.item)
-
-      invalidateSectionMaxYsCacheForSectionIndices(startingAt: indexPath.section)
```

Comment:
Note that we now only invalidate the section max Y cache when we're sizing something "after batch updates" (this function can be invoked with index paths for old and current views of the world, so it's possible an index path for an old, out of bounds item is passed in... that's the whole reason we have the `preferredHeightUpdateContextForPreferredHeightUpdateToItem` in here).

I suspect this was causing bugs previously, and might explain a few internal crashes we've seen.